include/fmt/ostream.h

```diff
@@ -81,24 +85,50 @@ template class filebuf_access<filebuf_access_tag,
                               decltype(&filebuf_type::_Myfile),
                               &filebuf_type::_Myfile>;
 
-inline bool write(std::filebuf& buf, fmt::string_view data) {
-  FILE* f = get_file(buf);
-  if (!f) return false;
-  print(f, data);
-  return true;
+inline bool write_ostream_msvc_unicode(std::ostream& os,
+                                       fmt::string_view data) {
+#ifdef _WIN32
+  if (auto* buf = dynamic_cast<std::filebuf*>(os.rdbuf()))
+    if (FILE* f = get_file(*buf)) return write_console_on_windows(f, data);
+#endif
+  return false;
+}
+inline bool write_ostream_msvc_unicode(std::wostream&,
+                                       fmt::basic_string_view<wchar_t>) {
+  return false;
+}
+
+#if defined(_WIN32) && defined(__GLIBCXX__)
+inline bool write_ostream_gcc_mingw_unicode(std::ostream& os,
+                                            fmt::string_view data) {
+  auto* rdbuf = os.rdbuf();
+  FILE* c_file;
+  if (auto* fbuf = dynamic_cast<__gnu_cxx::stdio_sync_filebuf<char>*>(rdbuf))
+    c_file = fbuf->file();
+  else if (auto* fbuf = dynamic_cast<__gnu_cxx::stdio_filebuf<char>*>(rdbuf))
+    c_file = fbuf->file();
+  else
+    return false;
+  if (c_file) return write_console_on_windows(c_file, data);
+  return false;
 }
-inline bool write(std::wfilebuf&, fmt::basic_string_view<wchar_t>) {
+
+inline bool write_ostream_gcc_mingw_unicode(std::wostream&,
+                                            fmt::basic_string_view<wchar_t>) {
   return false;
 }
+#endif
 
 // Write the content of buf to os.
 // It is a separate function rather than a part of vprint to simplify testing.
 template <typename Char>
 void write_buffer(std::basic_ostream<Char>& os, buffer<Char>& buf) {
-  if (const_check(FMT_MSC_VERSION)) {
-    auto filebuf = dynamic_cast<std::basic_filebuf<Char>*>(os.rdbuf());
-    if (filebuf && write(*filebuf, {buf.data(), buf.size()})) return;
-  }
+  // TODO: check detail::is_utf8(). Here or bellow in print or in vprint?
```

Comment:
We don't use TODOs so let's remove this. A separate PR to add a check would be welcome.

Reply:
Fixed